src/interpreter/Interpreter.cpp

```diff
@@ -686,6 +711,22 @@ ByteCodeStackOffset* Interpreter::interpret(ExecutionState& state,
         NEXT_INSTRUCTION();                                           \
     }
 
+#define SIMD_MEMORY_LOAD_SPLAT_MEMIDX_OPERATION(opcodeName, opType)              \
+    DEFINE_OPCODE(opcodeName)                                                    \
+        :                                                                        \
+    {                                                                            \
+        using Type = typename SIMDType<opType>::Type;                            \
+        MemoryLoadMemidx* code = (MemoryLoadMemidx*)programCounter;              \
+        uint32_t offset = readValue<uint32_t>(bp, code->srcOffset());            \
+        opType value;                                                            \
+        memories[code->memIndex()]->load(state, offset, code->offset(), &value); \
+        Type result;                                                             \
```

Comment:
I would group the type definitions. One statement, one definition looks strange to me.